view/frontend/web/js/components/validators/core.js

```diff
@@ -65,6 +65,18 @@
             },
             $t('Please select one of the options.')
         ],
+        'validate-no-utf8mb4-characters': [
+            (value) => !/(?:[\uD800-\uDBFF][\uDC00-\uDFFF])/g.test(value),
+            (value) => {
+                var message = $t('Please remove invalid characters: {0}.'),
```

Comment:
Thank you!

Since the function with error text is called when validation is failed, we can remove the checks:

```js
(value) => {
    var matches = value.match(/(?:[\uD800-\uDBFF][\uDC00-\uDFFF])/g);
    return $t('Please remove invalid characters: {0}.').replace('{0}', matches.join());
}
```